src/pudl/transform/sec10k.py

```diff
@@ -0,0 +1,471 @@
+        company_name=lambda x: x["company_name"].str.lower(),
+        sec10k_type=lambda x: x["sec10k_type"].str.lower(),
+    )
+    return filings
+
+
+@dg.multi_asset(
+    outs={
+        "_core_sec10k__quarterly_company_information": dg.AssetOut(
+            group_name="core_sec10k",
+        ),
+        "_core_sec10k__changelog_company_name": dg.AssetOut(
+            group_name="core_sec10k",
+        ),
+    }
+)
+def core_sec10k__company_info(
+    raw_sec10k__quarterly_company_information: pd.DataFrame,
+):
+    """Reshape data from the raw SEC 10-K company information table.
+
+    Each SEC 10-K filing contains a header block that lists information the filer, and
+    potentially a number of other related companies. Each company level block of
+    information may include the company's name, physical address, some details about
+    how it files the SEC 10-K, and some other company attributes like taxpayer ID
+    number, industrial classification, etc. It may also include a mailing address that
+    is distinct from the physical address, and the history of names that the company has
+    had over time.
+
+    In its original form, this data is mostly contained in two poorly normalized "key"
+    and "value" columns, with the key indicating what kind of data will be contained in
+    the value column. Here we pivot the keys into column names so that each column has
+    a homogeneous type of value.
+
+    Each filing may contain information about multiple companies, with each identified
+    by a different filer_count. For each company there are several different potential
+    types of information provide in distinct named "blocks" (business_address,
+    company_data, filing_values, mail_address, and former_company). Each block type
+    needs to be extracted and pivoted independently to avoid column name collisions and
+    because they don't all have the same set of block_count and filer_count values.
+
+    After they've been pivoted, the blocks are merged together into two disinct tables:
+
+    - per-filing company and filing information
+    - the history of company name changes
+
+    This information is used elsewhere for matching SEC 10-K filers to EIA utilities
+    and for identifying the companies that are mentioned in the unstructured Exhibit 21
+    attachements.
+
+    There is a 1-to-1 correspondence between the company_data and filing_values blocks,
+    and virtually all business_address blocks (99.9%), but a substantial number (~5%) of
+    mail_address blocks do not share index values with any company_data or filing_values
+    blocks, and are ultimately dropped. The former_company blocks contain a large amount
+    of duplicative data, but can be associated with a central_index_key value for later
+    association with an individual SEC 10-K filer.
+
+    Further processing that standardizes the contents of these tables is deferred to
+    separate downstream core assets.
+    """
```

Comment:
Not sure if some of this detail should go into the table description. It seemed a bit too far in the weeds.

Reply:
Hopefully the `block_count` fix upstream in the Mozilla repo will obviate most of this because all the blocks will just line up nice and clean.